plugin/rewrite/reverter_test.go

```diff
@@ -0,0 +1,55 @@
+package rewrite
+
+import (
+	"testing"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/coredns/coredns/plugin/pkg/dnstest"
+	"github.com/coredns/coredns/plugin/test"
+
+	"github.com/miekg/dns"
+	"golang.org/x/net/context"
+)
+
+func TestResponseReverter(t *testing.T) {
+	rules := []Rule{}
+	r, _ := newNameRule("stop", "regex", `(core)\.(dns)\.(rocks)`, "{2}.{1}.{3}", "answer", "name", `(dns)\.(core)\.(rocks)`, "{2}.{1}.{3}")
+	rules = append(rules, r)
+
+	tests := []struct {
+		from     string
+		fromType uint16
+		answer   []dns.RR
```

Comment:
this doesn't need to be a slice, we only ever put 1 RR in it, did you intent to test multiple?

Reply:
> did you intent to test multiple?

@miekg , yes, I did. Good point. I will add that test case.

As a side note, I was wondering about multiple names in a single request. I don't think the plugin supports it at the moment.

Reply:
there can only be one name in the question section in DNS

Reply:
>  there can only be one name in the question section in DNS

@miekg , I see 👍 . I was self-educating myself the other day and stumbled upon [stack thread](
https://stackoverflow.com/questions/4082081/requesting-a-and-aaaa-records-in-single-dns-query/4083071#4083071). Then, kept it in memory to ask someone about it 😃